R/seurat.R

```diff
@@ -2465,13 +2474,20 @@ SetAssayData.Seurat <- function(
   ...
 ) {
   CheckDots(...)
+  if (is_present(arg = slot)) {
+    .Deprecate(
+      when = '5.0.0',
+      what = 'GetAssayData(slot = )',
+      with = 'GetAssayData(layer = )'
+    )
+    layer <- slot
+  }
   object <- UpdateSlots(object = object)
   assay <- assay %||% DefaultAssay(object = object)
   object[[assay]] <- SetAssayData(
     object = object[[assay]],
     layer = layer,
     new.data = new.data,
-    slot = slot,
```

Comment:
This now fails
pbmc3k <- LoadData('pbmc3k')
pbmc3k <- SetAssayData(
    object = pbmc3k,
    assay = "RNA",
    slot = "data",
    new.data = pbmc3k[["RNA"]]$data
)